client/asset/eth/eth.go

```diff
@@ -4065,69 +3976,79 @@ func (w *assetWallet) checkPendingApprovals() {
 	}
 }
 
+func (w *baseWallet) unconfirmedTxsCopy() map[uint64]*extendedWalletTx {
+	w.unconfirmedTxsMtx.Lock()
+	defer w.unconfirmedTxsMtx.Unlock()
+
+	unconfirmed := make(map[uint64]*extendedWalletTx, len(w.unconfirmedTxs))
+	for nonce, tx := range w.unconfirmedTxs {
+		unconfirmed[nonce] = tx
+	}
+
+	return unconfirmed
+}
+
 // sumPendingTxs sums the expected incoming and outgoing values in unconfirmed
-// transactions stored in pendingTxs. Not used if the node is a txPoolFetcher.
+// transactions stored in unconfirmedTxs. Not used if the node is a
+// txPoolFetcher.
 func (w *assetWallet) sumPendingTxs(bal *big.Int) (out, in uint64) {
-	w.tipMtx.RLock()
-	tip := w.currentTip.Number.Uint64()
-	w.tipMtx.RUnlock()
-
 	isToken := w.assetID != w.baseChainID
+	balanceHasChanged := w.pendingTxCheckBal == nil || bal.Cmp(w.pendingTxCheckBal) != 0
+	w.pendingTxCheckBal = bal
```

Comment:
You're right, this field was no longer being protected. `pendingTxMtx`/`pendingTxs' existed previously. At one point in development I had both `unconfirmedTxs` and `pendingTxs`, then I got rid of `pendingTxs`. I'm going to rename `unconfirmedTxs` to `pendingTxs` since the name is more compact, and I'll protect this field with the mutex.